test/test_prims.py

```diff
@@ -310,6 +310,15 @@ def func(a):
         self.assertFalse(includes_prims_digamma)
         self.assertTrue(includes_nvprims_exp)
 
+    def test_tensor_to_overload_to_refs(self, device):
```

Comment:
Let's use OpInfo for tests instead, then this would be tested in `test_ops.py`. We need to uncomment this:
https://github.com/pytorch/pytorch/blob/c5a8946e40d6cda42aa38dda2705ea4e9930c2cb/torch/testing/_internal/common_methods_invocations.py#L15984-L15989

Reply:
Thx for suggesting the opinfo test. Turns out my python overload is buggy :smile:

Reply:
I'm actually also able to repro the data dependent control flow issue~~ double win~

Reply:
oh... the control flow issue went away after this
```
-            or a.is_contiguous(memory_format=memory_format)
+            or utils.is_contiguous_for_memory_format(a, memory_format=memory_format)
```